components/credentials/src/main/kotlin/org/cloudfoundry/credhub/credential/CertificateCredentialValue.kt

```diff
@@ -55,6 +55,9 @@ class CertificateCredentialValue : CredentialValue {
     @JsonIgnore
     var versionCreatedAt: Instant? = null
 
+    @JsonInclude(Include.NON_NULL)
```

Comment:
I'm a bit confused by the `NON_NULL` tag together with the nullable value. Could you elaborate please?

Reply:
This is a [jackson annotation](https://www.baeldung.com/jackson-ignore-null-fields) that means: if this field has null value, do not include it when serializing this java object into json. Aka, ignore when null.

Reply:
I don't understand what difference that makes then.

If you include null values, then the variable gets set to null.
If you don't, the variable stays set to null.

Doesn't it equal null regardless?

Reply:
This is just about how the resulting serialized json looks like. For example, when you have the java object `foo=abc, bar=null`:
- with the NON_NULL annotation, the resulting json looks like `{foo:abc}`
- without the NON_NULL annotation, the resulting json looks like `{foo:abc, bar:null}`

Reply:
I see. So I guess my main point, following the discussion about this field being nullable, is that if this field can't be null, we don't need any shady annotation.

Reply:
Since the field is no longer nullable, it doesn't use this annotation anymore. 

However, since this class is used in contexts unrelated to the generate and regenerate endpoints, we have marked it instead with the `JsonIgnore` annotation. This directs the JSON renderer not to output the field at all. 

This was necessary since this class is also used in GET endpoints like `/api/v1/data?name=cert-name`, we do not want the field to appear at all - the duration being overridden (nor the originally requested duration) is not persisted data in the database per the requirements laid out in the original issue.

Reply:
I am confused again. Why did the "nullable" change impact the fact that the field should be included or not in JSON, and hence the `JsonIgnore` annotation?

Reply:
We only wanted to conditionally include the field in responses for the generate / regenerate API endpoints. However, this class is used also with the GET endpoints. When the field was nullable, it only ever get set to a non-null value when generating or regenerating a certificate. Due to the previous `JsonInclude(Include.NON_NULL)` annotation, this meant it was not rendered for the GET endpoints, since the value was null on those code paths.

However, since it is now not nullable, it will always have a value (`false` for `duration_overridden`, `0` for `duration_used`). This means that the GET endpoints would start rendering:
```json
{ "name": "some-cert", ...more properties..., "duration_overridden": false, "duration_used": 0 }
```

which is not desirable - these fields only apply when generating/regenerating the certificate. So we added the `JsonIgnore` to ensure it is not output. This is now controlled by the new `CertificateGenerationView` class introduced in one of the latest commits. The existing `trustedCa` and `versionCreatedAt` fields on this `CertificateCredentialValue` class have similar `JsonIgnore` annotations, likely for the same reason.